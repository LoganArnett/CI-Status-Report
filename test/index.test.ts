import { Application } from 'probot'

import ciStatusReport from '../src'

const basePayload = {
  installation: { id: 123 },
  repository: {
    name: 'CI-Status-Report-Test',
    owner: { login: 'LoganArnett' }
  }
}

describe('CI-Status-Report Tests', () => {
  let probot: any, github: any

  beforeEach(() => {
    probot = new Application()
    github = {
      issues: { createComment: jest.fn() }
    }

    probot.auth = jest.fn(() => Promise.resolve(github))
    ciStatusReport(probot)
  })

  it('should not create a comment when the state is anything but "failure"', async () => {
    const event = {
      name: 'status',
      payload: {
        ...basePayload,
        state: 'passed',
        context: 'ci/circleci'
      }
    }

    await probot.receive(event)
    expect(github.issues.createComment).not.toHaveBeenCalled()
  })

  it('should not create a comment for a context that it does not support', async () => {
    const event = {
      name: 'status',
      payload: {
        ...basePayload,
        state: 'failure',
        context: 'unsupported/context/provider'
      }
    }

    await probot.receive(event)
    expect(github.issues.createComment).not.toHaveBeenCalled()
  })

  it('should create a comment', async () => {
    const event = {
      name: 'status',
      payload: {
        ...basePayload,
        state: 'failure',
        context: 'ci/circleci'
      }
    }

    await probot.receive(event)
    expect(github.issues.createComment).toHaveBeenCalled()
  })
})
