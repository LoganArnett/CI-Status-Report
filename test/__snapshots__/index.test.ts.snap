// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CI-Status-Report Tests Circle CI Tests should create a PR comment for a failed build 1`] = `
"
<p align=\\"center\\">
<img width=\\"400\\" height=\\"125\\" src=\\"http://cloud.loganarnett.com/f3467142ee70/Image%2525202019-11-08%252520at%2525202.31.28%252520AM.png\\">
</p>


## Build [#7](https://circleci.com/gh/LoganArnett/CI-Status-Report-Test/7?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) Failed


This PR is moving right along but the build has failed - [#7](https://circleci.com/gh/LoganArnett/CI-Status-Report-Test/7?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).


The commit that started this build can be found here [**6b50c5d**](https://localhost:3000).


\`\`\`bash
yarn run v1.5.1
$ mocha


  Test Suite
    stringAddition helper
      1) should add strings and return a number


  0 passing (6ms)
  1 failing

  1) Test Suite
       stringAddition helper
         should add strings and return a number:

      AssertionError [ERR_ASSERTION]: 4 == 3
      + expected - actual

      -4
      +3
      
      at Context.it (test/index.test.js:8:14)



error An unexpected error occurred: \\"Command failed.
Exit code: 1
Command: sh
Arguments: -c mocha
Directory: /home/circleci/repo
Output:
\\".
info If you think this is a bug, please open a bug report with the information provided in \\"/home/circleci/repo/yarn-error.log\\".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
\`\`\`
Once you have updated the PR and the build is passing then :shipit:


"
`;

exports[`CI-Status-Report Tests Travis CI Tests should create a PR comment for a failed build 1`] = `
"
<p align=\\"center\\">
<img width=\\"400\\" height=\\"150\\" src=\\"http://cloud.loganarnett.com/be61972f4407/Image%2525202019-11-08%252520at%2525202.31.55%252520AM.png\\">
</p>


## Build [#3](https://travis-ci.org/LoganArnett/ci-status-report/builds/3?utm_source=github_status&utm_medium=notification) Failed


This PR is moving right along but the build has failed - [#3](https://travis-ci.org/LoganArnett/ci-status-report/builds/3?utm_source=github_status&utm_medium=notification).


The commit that started this build can be found here [**6b50c5d**](https://localhost:3000).


Once you have updated the PR and the build is passing then :shipit:


"
`;
